packages/commonwealth/server/routes/updateNewProfile.ts

```diff
@@ -0,0 +1,98 @@
+import type { Request, Response, NextFunction } from 'express';
+import type { DB } from '../models';
+
+export const Errors = {
+  NotAuthorized: 'Not authorized',
+  InvalidUpdate: 'Invalid update',
+  NoProfileFound: 'No profile found',
+  NoAddressFound: 'No address found',
+  NoAddressProvided: 'No address provided in query',
+};
+
+const updateNewProfile = async (
+  models: DB,
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  if (!req.body.address) {
+    return next(new Error(Errors.NoAddressProvided));
+  }
+
+  if (
+    !req.body.email &&
```

Comment:
We should use TypedRequest and TypedResponse here as well to be clear about the interface this is expecting.